packages/@react-spectrum/picker/src/Picker.tsx

```diff
@@ -134,6 +134,20 @@ function Picker<T extends object>(props: SpectrumPickerProps<T>, ref: DOMRef<HTM
     }
   }, [scale, isMobile, triggerRef, state.selectedKey]);
 
+  // Make sure we only unwrap if the trigger changes otherwise we'll retrigger
+  // the observer too often
+  let resizeRef = useMemo(() => unwrapDOMRef(triggerRef), [triggerRef]);
+
+  let onResize = useCallback(() => {
+    let width = triggerRef.current.UNSAFE_getDOMNode().offsetWidth;
+    setButtonWidth(width);
+  }, [triggerRef, setButtonWidth]);
```

Comment:
```suggestion
  let onResize = useCallback(() => {
    let width = resizeRef.current.offsetWidth;
    setButtonWidth(width);
  }, [resizeRef, setButtonWidth]);
```

Reply:
there are a few more lines where triggerRef is used and who could now use resizeRef but I did not change them do you think that would be a reasonable cleanup task for a follow up issue?